it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+                        .setStream(Stream
+                                .newBuilder()
+                                .setSourceConfig(buildSourceConfig(sourceConnectionProfileId))
+                                .setDisplayName(streamId)
+                                .setDestinationConfig(buildDestinationConfig(destinationConnectionProfileId))
+                                .setBackfillAll(Stream.BackfillAllStrategy.getDefaultInstance())
+                                .build())
+                        .build();
+
+        Stream reference = datastreamClient.createStreamAsync(request).get();
+        createdStreamIds.add(streamId);
+
+        return reference;
+    }
+
+    public Stream updateStreamState(String streamId, Stream.State state)
+            throws ExecutionException, InterruptedException {
+        Stream.Builder streamBuilder = Stream.newBuilder()
+                .setName(StreamName.format(projectId, location, streamId))
+                .setState(state);
+
+        FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+        UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                .setStream(streamBuilder)
+                .setUpdateMask(fieldMaskBuilder)
+                .build();
+
+        return datastreamClient.updateStreamAsync(request).get();
+    }
+
+    @Override
+    public Stream startStream(String streamId) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamId, Stream.State.RUNNING);
+    }
+
+    @Override
+    public Stream pauseStream(String streamID) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamID, Stream.State.PAUSED);
+    }
+
+    @Override
+    public void cleanUpAll() throws ExecutionException, InterruptedException {
+
+        for (String stream : createdStreamIds) {
+            datastreamClient.deleteStreamAsync(DeleteStreamRequest
+                    .newBuilder()
+                    .setName(StreamName.format(projectId, location, stream))
+                    .build()).get();
+        }
+
+        for (String connectionProfile : createdConnectionProfileIds) {
+            datastreamClient.deleteConnectionProfileAsync(DeleteConnectionProfileRequest
+                    .newBuilder().setName(ConnectionProfileName.format(projectId, location, connectionProfile))
+                    .build()).get();
+        }
+    }
+
+
+    /**
+     * Builder for {@link DefaultDatastreamResourceManager}.
+     */
+    public static final class Builder {
+        private final String projectId;
+        private final String location;
+        private DatastreamClient datastreamClient;
+        private Map<String, ArrayList<String>> allowedTables;
+        private String gcsBucketName;
+        private String gcsRootPath;
+        private DefaultDatastreamResourceManager.Source source;
+        private DefaultDatastreamResourceManager.Destination destination;
```

Comment:
Like I said in my previous comment on the constructor, I think most of these can be removed and refactored to be parameters to their respective methods.

Reply:
done